crates/primitives/src/l1.rs

```diff
@@ -352,6 +352,28 @@ impl BorshDeserialize for BitcoinAddress {
     }
 }
 
+impl PartialEq for BitcoinAddress {
+    fn eq(&self, other: &Self) -> bool {
+        self.address
+            .script_pubkey()
+            .eq(&other.address.script_pubkey())
+    }
+}
+
+impl PartialOrd for BitcoinAddress {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for BitcoinAddress {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.address
+            .script_pubkey()
+            .cmp(&other.address.script_pubkey())
+    }
+}
```

Comment:
The goal here was to compare the script pubkey, and that we won't be using addresses with different networks. Ideally, BitcoinAddress shouldn't even have a network field and should wrap a network-checked address, but that's something for later.

There is also no natural ordering of Network. So this should be fine imo.

Reply:
Then it shouldn't implement `Ord`.  The `Eq` impl is also wrong since it doesn't check all fields.  Either remove the ord/eq impls or make them also check the network field.